src/components/datepicker/datePicker.js

```diff
@@ -603,10 +913,23 @@
   /**
    * Sets whether the input is currently focused.
    * @param {boolean} isFocused
+   * @param {boolean} isBeginDate
    */
-  DatePickerCtrl.prototype.setFocused = function(isFocused) {
-    if (!isFocused) {
-      this.ngModelCtrl.$setTouched();
+  DatePickerCtrl.prototype.setFocused = function(isFocused, isBeginDate) {
```

Comment:
Have a function where the signature is `(boolean boolean)` is generally something you want to avoid. Can we break this up into two functions?